packages/core/protocols/src/proto/dxos/client/services.proto

```diff
@@ -292,8 +297,39 @@ message Invitation {
 
     /// Guest should call `Authenticate` with the shared secret.
     SHARED_SECRET = 1;
+
+    /// Invitation is only claimable by a specific identity.
+    IDENTITY = 2;
   }
 
+  /// Randomly id generated by the host.
+  string invitation_id = 1;
+
+  /// Determines the behavior of the invitation.
+  Type type = 2;
+
+  /// Kind of access the invitation will grant.
+  Kind kind = 3;
+
+  /// How the invitation is authenticated.
+  AuthMethod auth_method = 4;
+
+  /// Beacon allows two peers to rendezvous on a p2p network.
+  Beacon beacon = 5;
+
+  optional google.protobuf.Timestamp expiry = 6;
+
+  /// Space to join (only present if kind is SPACE).
+  optional dxos.keys.PublicKey space_key = 7;
+
+  /// Idenitity key of the invitation recipient (only present if auth_method is IDENTITY).
+  optional dxos.keys.PublicKey recipient = 8;
```

Comment:
I'd update the comments here to indicate that these are only filled out after the rendezvous, right? Currently they feel misleading that the space key would always be there if it's a space invitation, etc.